modules/@apostrophecms/piece-type/ui/apos/components/AposImageCropper.vue

```diff
@@ -59,42 +83,268 @@ export default {
+
+      focalPoint.style.cursor = 'grab';
+      focalPoint.style.transitionDuration = '0.1s';
+
+      document.removeEventListener('mousemove', this.onFocalPointMouseMove);
+      document.removeEventListener('mouseup', this.onFocalPointMouseUp);
+    },
+    /**
+     * Place focal point at the position where the image has been clicked
+     * and update its new coordinates.
+     * Position it in relation of the current target offset
+     * in order to set the `left` and `top` properties accordingly.
+     * Do not place focal point when cropper is changing (move or resize)
+     * to keep a friendly user experience.
+     */
+    onImageClick (event) {
+      if (this.isChangingCropper) {
+        return;
+      }
+
+      const { focalPoint } = this.$refs;
+
+      const focalPointSize = this.getFocalPointSize();
+
+      const left = event.clientX - event.currentTarget.offsetLeft - event.currentTarget.offsetParent.offsetLeft - focalPointSize.halfWidth;
+      const top = event.clientY - event.currentTarget.offsetTop - event.currentTarget.offsetParent.offsetTop - focalPointSize.halfHeight;
+
+      focalPoint.style.left = `${left}px`;
+      focalPoint.style.top = `${top}px`;
+
+      this.updateFocalPointCoordinatesDebounced();
+    },
+    /**
+     * Update stencil and focal point coordinates
+     * after cropper has changed and emit its new coordinates.
+     */
+    handleCropperChange (coordinates) {
+      this.isChangingCropper = false;
+
       if (
-        !this.isUpdatingCoordinates &&
-        this.checkCoordinatesDiff(coordinates, this.docFields.data)
+        !this.isUpdatingCropperCoordinates &&
+        this.checkCropperCoordinatesDiff(coordinates, this.docFields.data)
       ) {
+        this.storeStencilCoordinates();
+        this.updateFocalPointCoordinates();
+
         this.$emit('change', coordinates, false);
       }
 
-      this.isUpdatingCoordinates = false;
+      this.isUpdatingCropperCoordinates = false;
     },
-    checkCoordinatesDiff (coordinates, dataFields) {
-      return Object.entries(coordinates)
+    /**
+     * Return wether the cropper coordinates have changed or not.
+     */
+    checkCropperCoordinatesDiff (coordinates, dataFields) {
+      return Object
+        .entries(coordinates)
         .some(([ name, value ]) => dataFields[name] !== value);
+    },
+    /**
+     * Store the stencil actual coordinates relative to the viewport,
+     * which are used for focal point DOM manipulation and its
+     * relative position calculation.
+     */
+    storeStencilCoordinates () {
+      const stencilElement = document.querySelector('[data-stencil]');
+      const stencilStyle = window.getComputedStyle(stencilElement);
+
+      // TODO: DOMMatrixReadOnly is not IE compatible. Should we use something compatible with IE?
+      // In that case, use regular elem.style.transform and extract values with a regex
+      const matrix = new window.DOMMatrixReadOnly(stencilStyle.transform);
```

Comment:
@boutell do we handle Internet Explorer 11, or do we start from Edge?

Reply:
Editing support is modern browsers only (no IE11 support for user scene).